runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -1537,6 +1546,16 @@ class INTERPRETER_CLASS
 		J9VMJAVALANGVIRTUALTHREAD_SET_STATE(_currentThread, _currentThread->threadObject, newThreadState);
 
 		if (JAVA_LANG_VIRTUALTHREAD_BLOCKING == newThreadState) {
+			if (J9_EVENT_IS_HOOKED(_vm->hookInterface, J9HOOK_VM_MONITOR_CONTENDED_ENTER)
+				&& J9_ARE_NO_BITS_SET(continuation->runtimeFlags, J9VM_CONTINUATION_RUNTIMEFLAG_JVMTI_CONTENDED_MONITOR_ENTER_RECORDED)
+			) {
+				ALWAYS_TRIGGER_J9HOOK_VM_MONITOR_CONTENDED_ENTER(_vm->hookInterface, _currentThread, continuation->objectWaitMonitor->monitor);
```

Comment:
Compared the platform thread path, we are missing a `VM_VMAccess::setPublicFlags(currentThread, J9_PUBLIC_FLAGS_THREAD_BLOCKED)` call here

As `JAVA_LANG_VIRTUALTHREAD_BLOCKING` is set on the virtual thread, the java level call to `vthread.getState()` will work, but I believe `jvmtiGetThreadState()` will still return `RUNNABLE` for this vthread in the callback triggered. Also `jvmtiGetCurrentContendedMonitor()`  will not work inside this callback either.

Reply:
> Compared the platform thread path, we are missing a VM_VMAccess::setPublicFlags(currentThread, J9_PUBLIC_FLAGS_THREAD_BLOCKED) call here

I believe it should be done under a different PR if this line needs to be added outside of `if (J9_EVENT_IS_HOOKED(_vm->hookInterface, J9HOOK_VM_MONITOR_CONTENDED_ENTER)`

Reply:
The flag need to be set before the hooked call so the user code would be able see the blocked state set on the thread via `jvmtiGetThreadState()`, it also need to be cleared after as the blocked virtual thread unmounts so carrier thread's state doesn't stay as blocked.